### ISSUE TITLE
babel/semi does not recognize private properties

### ISSUE THREAD

**Issue by simonkotwicz** [babel/eslint-plugin-babel#193]
```
class Hello {
  #world
}
```
this is not flagged by the rule
```
babel/semi: [error, always]
```

**Comment by JLHwung** [comment]
Turns out it suffices to add `ClassPrivateProperty` to https://github.com/babel/eslint-plugin-babel/blob/0f558b317eaca5b6b9b3706c2ce807eb2af3e46f/rules/semi.js#L73

PR is welcome.
